File: extract_doc.py
#!/usr/local/bin/python
import requests
import time
import json
import sys
from constants import *
from secrets import *
import base64

'''
extract structured data from PDFs under 4.5MB or that require under 30 seconds to process
'''

def extract_from_local_doc():  
  print(doc_local_path)
  try:
    with open(doc_local_path, 'rb') as pdf_file:
      pdf_bytes = pdf_file.read()
      pdf_file.close()
  except IOError as err:
    raise SystemExit(err)
  url = "https://api.sensible.so/dev/extract/{}".format(doc_type)
  payload = pdf_bytes
  headers = {
  'Authorization': 'Bearer {}'.format(API_KEY),
  'Content-Type': 'application/pdf'
}

  try:
    response  = requests.request("POST", url, headers=headers, data=payload)
    response.raise_for_status()
  except requests.RequestException as err:
    print(response.text)
    raise SystemExit(err)
  print(json.dumps(response.json(), indent=2))
if __name__ == '__main__':
  extract_from_local_doc()

File: ruby_example/lib/extract_doc.rb
#!/usr/bin/ruby 
# frozen_string_literal: true

# TODO: delete require_relative?
#require_relative "ruby_example/version"
require "faraday"
require 'json'

# TODO: style problem that I don't use modules & classes & just define some constants?
# TODO: delete test vars on publish
# specify your variable values here 
Doc_type = "auto_insurance_quote"
Doc_local_path = "../../TODELETE_auto_insurance_anyco.pdf"
# TODO: on publish, specify API key inline here instead + delete test vars
# var API_KEY = "YOUR_API_KEY"
API_KEY = ENV['API_KEY']

def extract_from_local_doc()
  puts "Extracting from doc #{Doc_local_path}"
  begin 
    pdf_bytes = IO.binread(Doc_local_path)
  rescue IOError => e
    fail(e)
  end 
  url = URI("https://api.sensible.so/dev/extract/#{Doc_type}")
  response = Faraday.post(url) do |req|
    req.headers['Content-Type'] = 'application/pdf'
    req.body = pdf_bytes
    req.headers["Authorization"] = "Bearer #{API_KEY}"
  end 
  if !response.success?
    abort "The request failed: #{response.status} #{response.reason_phrase}"
  end  
  # TODO: is it a style problem that I don't tell Farady to expect response.body to be JSON?
  puts "EXTRACTED DATA"
  response_json = JSON.parse(response.body)
  puts JSON.pretty_generate(response_json) 
end

if __FILE__ == $PROGRAM_NAME
  # TODO: clean up error handling so I on check Doc_local_path 1x
  begin 
    size_Mb = File.size(Doc_local_path)/(1024*1024)
  rescue IOError => e
    fail(e)
  end 
  if size_Mb < 4.5
    extract_from_local_doc()
  end
end